_posts/en/newsletters/2019-04-30-newsletter.md

```diff
@@ -0,0 +1,203 @@
+  be relayed by the Bitcoin network. The current mitigation is for LN to
+  temporarily move such small payments to be a miner fee instead of an
+  output, depending on the game theory that if an attacker cannot steal
+  money, they won’t spend money to attack.
+
+- [How was the dust limit of 546 satoshis was chosen? Why not 550
+  satoshis?]({{bse}}86068) The Bitcoin Core transaction relay policy
+  sets a dust limit of 546 satoshis as the minimum amount for an output,
+  which seems a peculiar amount. Raghav Sood describes how 546 is three
+  times the minimum cost to create and spend a P2PKH output. A reference
+  is made to a [2013 discussion][dust convo].
+
+- [History of transactions in Lighting Network.]({{bse}}85901) A LN
+  beginner asks how the history of transactions conducted on LN for a
+  user can be saved and how a payer receives a proof of payment. Mark H
+  responds saying the LN wallet would need to save transaction history
+  for a user and provides a nice explanation of how a payment hash
+  provided to a payer results in a payment preimage reveal that serves
+  as proof of payment.
+
+- [How can my private key be revealed if I use the same nonce while
+  generating the signature?]({{bse}}85638) Pieter Wuille provides a
+  thorough answer that, if you’re familiar with the math used in
+  public key cryptography, demonstrates how a private
+  key is revealed in such circumstances.
+
+- [Why do lightning invoices expire?]({{bse}}85981) Rene Pickhardt
+  guesses that the primary reason would be a high
+  volume recipient with relatively low storage capability could run out
+  of storage or memory. An additional reason given is to provide some closure to
+  proceed if a payment is not initiated or completed rather than leave
+  it dangling. David Harding points out that traditional businesses put
+  expiration dates on invoices to avoid an obligation to deliver goods in the
+  future at a previously offered price.
+
+- [Are there still miners or mining pools which refuse to implement
+  SegWit?]({{bse}}86208) Mark Erhardt provides extensive analysis demonstrating
+  that essentially the answer is "no." Only 0.03% non-empty blocks from
+  the past year had no SegWit transactions, and the two primary miners of
+  those few blocks have demonstrated in 2019 that they’re mining blocks
+  with SegWit transactions.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].  Note: unless otherwise noted, all merges described
+for Bitcoin Core are to its master development branch; some may also be
+backported to its pending release.*
+
+- [Bitcoin Core #14039][] causes Bitcoin Core to reject transactions
+  that use the segwit-style extended transaction encoding when they
```

Comment:
I think it'd be good to say "transactions received over the P2P network or RPC" here

Reply:
I'm not opposed to this change, but I'd like to better understand its motivation.  How else does the software receive transactions?  I think you might be trying to imply that the change only applies to loose transactions, but if I'm reading the code correctly, it'll also reject blocks that use extended format for legacy transactions.

Reply:
The motivation was that it wasn't immediately obvious to me whether this affected just p2p, just rpc or both. Explicitly saying "P2P or RPC" makes it clear that the change applies to both.

If you think it's better to leave this as it is now, that's also fine with me.